tests/unit/maze_transformer/training/zanj/test_zanj_ht_save_load.py

```diff
@@ -0,0 +1,104 @@
+from pathlib import Path
+
+import torch
+from muutils.zanj import ZANJ
+from muutils.zanj.torchutil import (
+    ConfigMismatchException,
+    assert_model_cfg_equality,
+    assert_model_exact_equality,
+)
+
+from maze_transformer.training.config import (
+    BaseGPTConfig,
+    ConfigHolder,
+    MazeDatasetConfig,
+    TrainConfig,
+    ZanjHookedTransformer,
+)
+
+ZANJ_MODEL_CFG: ConfigHolder = ConfigHolder(
+    train_cfg=TrainConfig(name="test_cfg_save-train"),
+    dataset_cfg=MazeDatasetConfig(name="test_cfg_save-data", grid_n=5, n_mazes=10),
+    model_cfg=BaseGPTConfig(
+        name="test_cfg_save-model",
+        act_fn="relu",  # need an actual function here since HookedTransformer will complain otherwise
+        d_model=128,
+        d_head=64,
+        n_layers=4,
+    ),
+)
+
+MODEL: ZanjHookedTransformer = ZanjHookedTransformer(ZANJ_MODEL_CFG)
+MODEL_C: ZanjHookedTransformer = ZANJ_MODEL_CFG.create_model_zanj()
+
+
+def _assert_model_output_equality(
+    model_a: ZanjHookedTransformer, model_b: ZanjHookedTransformer
+):
+    try:
+        assert_model_cfg_equality(model_a, model_b)
+    except ConfigMismatchException as e:
+        if e.diff == {
+            "model_cfg": {"are_weights_processed": {"self": False, "other": True}}
+        } or e.diff == {
+            "model_cfg": {
+                "are_layernorms_folded": {"self": False, "other": True},
```

Comment:
this is for testing saving/loading a model with processed weights/folded layernorm, where the state dict keys wont match. since we store this information in the config, we need to account for that part of the config being different